### PR TITLE
Add support for windows platforms

### PR DESCRIPTION
**Pull request by vsebe**
Signed-off-by: Violeta Sebe <vsebe@ca.ibm.com>

### CODE REVIEW

**Comment by vsebe** [comment]
Depends on https://github.com/eclipse/openj9/pull/829

**Comment by vsebe** [comment]
attn @pshipton

File: jdk/src/windows/bin/amd64/jvm.cfg [hunk 1]
```diff
@@ -34,5 +34,8 @@
 # "-XXaltjvm=<jvm_dir>" option, but that too is unsupported
 # and may not be available in a future release.
 #
--server KNOWN
--client IGNORE
+-j9vm KNOWN
+-hotspot IGNORE
```
Comment (pshipton) [inline jdk/src/windows/bin/amd64/jvm.cfg]: I suppose this just copied what is done for other platforms, but comparing with the previous values, it seems the ignores should be:
```
-server IGNORE
-client IGNORE
```

File: jdk/make/closed/autoconf/custom-hook.m4 [hunk 2]
```diff
@@ -110,7 +110,14 @@ AC_DEFUN_ONCE([OPENJ9_PLATFORM_SETUP],
...
 
   if test "x$OPENJ9_CPU" = xx86-64; then
-    OPENJ9_PLATFORM_CODE=xa64
+    if test "x$OPENJDK_BUILD_OS" = xlinux; then
+      OPENJ9_PLATFORM_CODE=xa64
+    elif test "x$OPENJDK_BUILD_OS" = xwindows; then
+      OPENJ9_PLATFORM_CODE=wa64
+      OPENJ9_BUILDSPEC="win_x86-64_cmprssptrs"
+    else
+      AC_MSG_ERROR([Unsupported OpenJ9 platform ${OPENJDK_BUILD_OS}, contact support team!])
```
Comment (pshipton) [inline jdk/make/closed/autoconf/custom-hook.m4]: Please remove ", contact support team!". The "support team" for OpenJ9 are the git repos, where an issue would be opened if appropriate.

File: jdk/make/closed/autoconf/generated-configure.sh [hunk 3]
```diff
@@ -8249,7 +8237,14 @@ $as_echo "$DEBUG_LEVEL" >&6; }
...
 
   if test "x$OPENJ9_CPU" = xx86-64; then
-    OPENJ9_PLATFORM_CODE=xa64
+    if test "x$OPENJDK_BUILD_OS" = xlinux; then
+      OPENJ9_PLATFORM_CODE=xa64
+    elif test "x$OPENJDK_BUILD_OS" = xwindows; then
+      OPENJ9_PLATFORM_CODE=wa64
+      OPENJ9_BUILDSPEC="win_x86-64_cmprssptrs"
+    else
+      as_fn_error $? "Unsupported OpenJ9 platform ${OPENJDK_BUILD_OS}, contact support team!" "$LINENO" 5
```
Comment (pshipton) [inline jdk/make/closed/autoconf/generated-configure.sh]: Please remove ", contact support team!". The "support team" for OpenJ9 are the git repos, where an issue would be opened if appropriate.

File: jdk/make/closed/autoconf/generated-configure.sh [hunk 4]
```diff
@@ -36813,7 +36813,11 @@ $as_echo "$OUTPUT_DIR_IS_LOCAL" >&6; }
 # At the end, call the custom hook. (Dummy macro if no custom sources available)
 
   # Add the J9VM vm lib directory into native LDFLAGS_JDKLIB path
-  LDFLAGS_JDKLIB="${LDFLAGS_JDKLIB} -L${JDK_OUTPUTDIR}/../vm"
+  if test "x$OPENJDK_BUILD_OS_ENV" = xwindows.cygwin; then
+    LDFLAGS_JDKLIB="${LDFLAGS_JDKLIB} -libpath:${JDK_OUTPUTDIR}/../vm/lib"
```
Comment (pshipton) [inline jdk/make/closed/autoconf/generated-configure.sh]: Not sure why the directory is different from other platforms.

File: closed/OpenJ9.gmk [hunk 5]
```diff
@@ -258,7 +271,10 @@ run-preprocessors-j9 : stage-j9 \
 
 build-j9 : run-preprocessors-j9
 	@$(ECHO) Compiling OpenJ9 in $(OUTPUT_ROOT)/vm
-	(export OMR_DIR=$(OUTPUT_ROOT)/vm/omr OPENJ9_BUILD=true && cd $(OUTPUT_ROOT)/vm && $(MAKE) $(MAKEFLAGS) JAVA_VERSION=80 VERSION_MAJOR=8 all)
+	(export OPENJ9_BUILD=true $(EXPORT_NO_USE_MINGW) $(EXPORT_MSVS_ENV_VARS) \
```
Comment (pshipton) [inline closed/OpenJ9.gmk]: Why is OMR_DIR removed?

File: closed/make/Main.gmk [hunk 6]
```diff
@@ -30,7 +30,7 @@ clean-j9vm:
 j9vm-build :
 	+($(CD) $(SRC_ROOT)/closed && $(MAKE) -f OpenJ9.gmk SPEC=$(SPEC) build-j9)
 
-j9vm-compose-buildjvm : j9vm-build $(J9JCL_GENSRC_MAKEFILE)
+j9vm-compose-buildjvm : j9vm-build
```
Comment (pshipton) [inline closed/make/Main.gmk]: Why is $(J9JCL_GENSRC_MAKEFILE) removed?

File: jdk/make/closed/autoconf/generated-configure.sh (hunk 4 above)
Comment (vsebe) [reply jdk/make/closed/autoconf/generated-configure.sh]: it is on win

File: closed/OpenJ9.gmk (hunk 5 above)
Comment (vsebe) [reply closed/OpenJ9.gmk]: Not required.

File: closed/make/Main.gmk (hunk 6 above)
Comment (vsebe) [reply closed/make/Main.gmk]: It does not exist.

File: jdk/src/windows/bin/amd64/jvm.cfg (hunk 1 above)
Comment (vsebe) [reply jdk/src/windows/bin/amd64/jvm.cfg]: Should [re]add them to all jvm.cfg files?

File: jdk/make/closed/autoconf/custom-hook.m4 (hunk 2 above)
Comment (vsebe) [reply jdk/make/closed/autoconf/custom-hook.m4]: done

**Comment by vsebe** [comment]
I fixed jvm.cfg, custom-hook.m4 and generated-configure.sh; ready for review, thx

File: jdk/make/closed/autoconf/generated-configure.sh (hunk 4 above)
Comment (pshipton) [reply jdk/make/closed/autoconf/generated-configure.sh]: I realize its on Windows, but what is it about Windows that makes the directory different?

File: jdk/src/windows/bin/amd64/jvm.cfg (hunk 1 above)
Comment (pshipton) [reply jdk/src/windows/bin/amd64/jvm.cfg]: I think that can be done separately from this commit.

File: jdk/src/windows/bin/amd64/jvm.cfg [hunk 7]
```diff
@@ -34,5 +34,10 @@
 # "-XXaltjvm=<jvm_dir>" option, but that too is unsupported
 # and may not be available in a future release.
 #
--server KNOWN
+-j9vm KNOWN
+-hotspot IGNORE
```
Comment (pshipton) [inline jdk/src/windows/bin/amd64/jvm.cfg]: You can remove the IGNORE for hotspot, classic, native, and green, since they didn't exist before.

Comment (vsebe) [reply jdk/src/windows/bin/amd64/jvm.cfg]: done

**Comment by vsebe** [comment]
@pshipton cleaned up jvm.cfg as suggested, thx.

File: jdk/make/closed/autoconf/generated-configure.sh (hunk 4 above)
Comment (pshipton) [reply jdk/make/closed/autoconf/generated-configure.sh]: Seems on Windows you link to the .lib files which are in the lib subdirectory, while on other platforms you link to the .so files which are in the root vm directory.

File: .gitignore [hunk 8]
```diff
@@ -0,0 +1,5 @@
+/.project
+build/
```
Comment (pshipton) [inline .gitignore]: To be consistent with the other directories and openj9-openjdk-jdk9, `build/` should start with a slash

**Pull request merged**